view/frontend/web/instantsearch.js

```diff
@@ -56,6 +56,16 @@ requirejs(['algoliaBundle'], function(algoliaBundle) {
 			urlSync: {
 				useHash: true,
 				trackedParameters: ['query', 'page', 'attribute:*', 'index']
+			},
+			searchFunction: function(helper) {
+				if (helper.state.query === '' && !algoliaConfig.isSearchPage) {
```

Comment:
How it works for category pages powered by Algolia? When it loads it has an empty query and is not a search page - wouldn't it break it?

Reply:
I think category page are considered search page. I worked when I tested but I will verify, just to make sure I didn't miss something.

Reply:
Any news on that @julienbourdeau ?

Reply:
Yes I confirm that `algoliaConfig.isSearchPage` is true for category pages. And only the pages that display results (not the top level page with special template).

Reply:
Thanks @julienbourdeau!